infrastructure/http_metadata_service_test.go

```diff
@@ -422,4 +422,70 @@ func describeHTTPMetadataService() {
 		})
 
 	})
+
+	Describe("GetServerName from encoded user data by IaaS", func() {
+		var (
+			ts         *httptest.Server
+			serverName *string
+		)
+
+		handlerFunc := func(w http.ResponseWriter, r *http.Request) {
+			defer GinkgoRecover()
+
+			Expect(r.Method).To(Equal("GET"))
+			Expect(r.URL.Path).To(Equal("/user-data"))
+			Expect(r.Header.Get("key")).To(Equal("value"))
+
+			var jsonStr string
+
+			if serverName == nil {
+				jsonStr = `Int9Ig==`
+			} else {
+				jsonStr = fmt.Sprint(`Insic2VydmVyIjp7Im5hbWUiOiJmYWtlLXNlcnZlci1uYW1lIn19Ig==`)
+			}
+
+			w.Write([]byte(jsonStr))
+		}
+
+		BeforeEach(func() {
+			serverName = nil
+
+			handler := http.HandlerFunc(handlerFunc)
+			ts = httptest.NewServer(handler)
+			metadataService = NewHTTPMetadataService(ts.URL, metadataHeaders, "/user-data", "/instanceid", "/ssh-keys", dnsResolver, platform, logger)
+		})
+
+		AfterEach(func() {
+			ts.Close()
+		})
+
+		Context("when the server name is present in the JSON", func() {
+			BeforeEach(func() {
+				name := "fake-server-name"
+				serverName = &name
+			})
+
+			It("returns the server name", func() {
+				name, err := metadataService.GetServerName()
+				Expect(err).ToNot(HaveOccurred())
+				Expect(name).To(Equal("fake-server-name"))
```

Comment:
how is this test succeeding? `Insic2VydmVyIjp7Im5hbWUiOiJmYWtlLXNlcnZlci1uYW1lIn19Ig==` start with `"` which means it would be interpreted as a string, whereas parsing logic expects a struct?